### PR TITLE
feat: allow to override generated rockcraft.yaml

### PR DESCRIPTION
**Pull request by vpa1977**
Generated rockcraft.yaml may not be the ideal option. This PR adds two new options:
- `rockcraftYaml` - specify override `rockcraft.yaml`
- `createService`  (default true) - whether to create service section.

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        HashMap<String, Object> map2 = new HashMap<String, Object>();
+        map1.put("b", "c");
+        // when maps are merged
+        Map<String, Object> ret = MapMerger.merge(map1, map2);
+        // then the result is (a->b, b->c)
+        assertEquals(2, ret.keySet().size());
+        assertEquals("b", ret.get("a"));
+        assertEquals("c", ret.get("b"));
+    }
+
```
Comment (pushkarnk) [inline rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java:53]: Not annotated as `@Test`.

File: rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        HashMap<String, Object> map2 = new HashMap<String, Object>();
+        map1.put("a", "d");
+        // when maps are merged
+        Map<String, Object> ret = MapMerger.merge(map1, map2);
+        // then the result is (a->d)
+        assertEquals(1, ret.keySet().size());
+        assertEquals("d", ret.get("a"));
+    }
+
+    public void testSubmapMerge() {
```
Comment (pushkarnk) [inline rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java:67]: This is also not annotated as `@Test`. This test might fail.

File: rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        inner.put("a", "a");
+        inner.put("b", "b");
+        inner.put("f", "f");
+        map1.put("a", inner);
+        HashMap<String, Object> map2 = new HashMap<String, Object>();
+        inner = new HashMap<String, Object>();
+        inner.put("a", "b");
+        inner.put("b", "c");
+        inner.put("d", "e");
+        map1.put("a", inner);
```
Comment (pushkarnk) [inline rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java:80]: Do we intend to do a `map2.put("a", inner)` here?

File: rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        // when maps are merged
+        Map<String, Object> ret = MapMerger.merge(map1, map2);
+        // then the result is (a->(a->b, b->c, d->e, f->f) )
+        assertEquals(1, ret.keySet().size());
+        ret = (Map<String, Object>) ret.get("a");
+        assertEquals(4, ret.keySet().size());
+        assertEquals("b", ret.get("a"));
+        assertEquals("b", ret.get("a"));
```
Comment (pushkarnk) [inline rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java:88]: Redundant.

**Review by pushkarnk — CHANGES_REQUESTED** [review]

File: rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java (hunk 3 above)
Comment (vpa1977) [reply rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java]: Thanks!!! indeed was a typo.

File: rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java (hunk 4 above)
Comment (vpa1977) [reply rockcraft/src/test/java/com/canonical/rockcraft/builder/MapMergerTest.java:90]: And another typo. Thanks !!!!

**Review by pushkarnk — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**